tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator.Console/ExtensionsMetadataGenerator.cs

```diff
@@ -183,5 +183,35 @@ public static string GetReflectionFullName(TypeReference typeRef)
         {
             return typeRef.FullName.Replace("/", "+");
         }
+
+        public class FunctionsAssemblyResolver : DefaultAssemblyResolver
+        {
+            private static HashSet<string> _trustedPlatformAssemblies = GetTrustedPlatformAssemblies();
+
+            private static HashSet<string> GetTrustedPlatformAssemblies()
+            {
+                var tpas = new HashSet<string>();
+                var data = AppContext.GetData("TRUSTED_PLATFORM_ASSEMBLIES");
+                foreach (var tpa in data.ToString().Split(";"))
+                {
+                    tpas.Add(tpa);
+                }
+                return tpas;
+            }
+
+            public override AssemblyDefinition Resolve(AssemblyNameReference name)
+            {
+                var assemblyDef = base.Resolve(name);
+
+                // As soon as we get to a TPA we can stop. This helps prevent the odd circular reference
+                // with type forwarders as well.
+                if (_trustedPlatformAssemblies.Contains(assemblyDef.MainModule.FileName))
```

Comment:
Should we worry about supporting case insensitive Contains check?

```
var tpas = new HashSet<string>(StringComparer.OrdinalIgnoreCase);

Reply:
yeah good call...